docs/release.rst

```diff
@@ -0,0 +1,30 @@
+***************
+Release Process
+***************
+
+As this project is critical to the Python ecosystem's supply chain security, all
+releases are PGP signed with one of the keys listed in the `installation page`_.
+Before releasing please make sure you PGP key is listed there, and preferably
+signed by one of the other key holders. If your key is not signed my one of the
+other key holders, please make sure the PR that added your key to the
+`installation page`_ was approved by at least one other maintainer.
+
+After that is done, you may release the project by following these steps:
+
+#. Bump the versions in ``setup.cfg`` and ``src/build/__init__.py``
```

Comment:
Reminder, needs to be updated after #471.

Reply:
I merged it now, I am gonna update this PR.